--- worker-install.sh.orig	2016-12-14 06:58:48.000000000 +0000
+++ worker-install.sh	2016-12-14 06:59:22.000000000 +0000
@@ -30,6 +30,9 @@
 # Determines the container runtime for kubernetes to use. Accepts 'docker' or 'rkt'.
 export CONTAINER_RUNTIME=docker
 
+# to mask update-engine ?
+export IS_MASK_UPDATE_ENGINE=true
+
 # The above settings can optionally be overridden using an environment file:
 ENV_FILE=/run/coreos-kubernetes/options.env
 
@@ -315,7 +318,9 @@
 
 chmod +x /opt/bin/host-rkt
 
-systemctl stop update-engine; systemctl mask update-engine
+if [ "$IS_MASK_UPDATE_ENGINE" = true ]; then
+        systemctl stop update-engine; systemctl mask update-engine
+fi
 
 systemctl daemon-reload
 
