--- worker-install.sh.orig	2016-12-14 06:50:11.000000000 +0000
+++ worker-install.sh	2016-12-14 06:50:04.000000000 +0000
@@ -4,6 +4,24 @@
 # List of etcd servers (http://ip:port), comma separated
 export ETCD_ENDPOINTS=
 
+# etcd scheme
+export ETCD_SCHEME="https"
+
+# etcd certificates root directory
+export ETCD_CERT_ROOT_DIR="/etc/ssl/etcd"
+
+# Should etcd tls certificates should be enabled to connect to etcd servers
+export ETCD_CLIENT_CERT_AUTH=true
+
+# etcd certificate file
+export ETCD_CERT_FILE="/etc/ssl/etcd/etcd2.pem"
+
+# etcd key file
+export ETCD_KEY_FILE="/etc/ssl/etcd/etcd2-key.pem"
+
+# etcd CA certificate
+export ETCD_TRUSTED_CA_FILE="/etc/ssl/etcd/ca.pem"
+
 # The endpoint the worker node should use to contact controller nodes (https://ip:port)
 # In HA configurations this should be an external DNS record or loadbalancer in front of the control nodes.
 # However, it is also possible to point directly to a single control node.
@@ -257,6 +275,9 @@
 FLANNELD_IFACE=$ADVERTISE_IP
 FLANNELD_ETCD_ENDPOINTS=$ETCD_ENDPOINTS
 EOF
+    if [ "$ETCD_CLIENT_CERT_AUTH" = true ]; then
+    printf "FLANNELD_ETCD_CAFILE=$ETCD_TRUSTED_CA_FILE\nFLANNELD_ETCD_CERTFILE=$ETCD_CERT_FILE\nFLANNELD_ETCD_KEYFILE=$ETCD_KEY_FILE\n" >> $TEMPLATE
+    fi
     fi
 
     local TEMPLATE=/etc/systemd/system/flanneld.service.d/40-ExecStartPre-symlink.conf.conf
@@ -306,6 +327,11 @@
     }
 }
 EOF
+        if [ "$ETCD_CLIENT_CERT_AUTH" = true ]; then
+       sed -i "7i \     \"etcd_key_file\": \"$ETCD_KEY_FILE\"," $TEMPLATE
+       sed -i "8i \     \"etcd_cert_file\": \"$ETCD_CERT_FILE\"," $TEMPLATE
+       sed -i "9i \     \"etcd_ca_cert_file\": \"$ETCD_TRUSTED_CA_FILE\"," $TEMPLATE
+        fi
     fi
 
 }
