--- worker-install.sh.orig	2016-12-09 20:09:28.000000000 +0000
+++ worker-install.sh	2016-12-09 20:45:36.000000000 +0000
@@ -4,6 +4,24 @@
 # List of etcd servers (http://ip:port), comma separated
 export ETCD_ENDPOINTS=
 
+# etcd scheme
+export ETCD_SCHEME="https"
+
+# etcd certificates root directory
+export ETCD_CERT_ROOT_DIR="/etc/ssl/etcd"
+
+# Should etcd tls certificates should be enabled to connect to etcd servers
+export ETCD_CLIENT_CERT_AUTH=true
+
+# etcd certificate file
+export ETCD_CERT_FILE="/etc/ssl/etcd/etcd2.pem"
+
+# etcd key file
+export ETCD_KEY_FILE="/etc/ssl/etcd/etcd2-key.pem"
+
+# etcd CA certificate
+export ETCD_TRUSTED_CA_FILE="/etc/ssl/etcd/ca.pem"
+
 # The endpoint the worker node should use to contact controller nodes (https://ip:port)
 # In HA configurations this should be an external DNS record or loadbalancer in front of the control nodes.
 # However, it is also possible to point directly to a single control node.
@@ -188,6 +206,14 @@
 [Install]
 WantedBy=multi-user.target
 EOF
+        if [ "$ETCD_CLIENT_CERT_AUTH" = true ]; then
+       sed -i "8i PermissionsStartOnly=true" $TEMPLATE	
+       sed -i "15i Environment=ETCD_CA_CERT_FILE=$ETCD_TRUSTED_CA_FILE" $TEMPLATE
+       sed -i "16i Environment=ETCD_CERT_FILE=$ETCD_CERT_FILE" $TEMPLATE
+       sed -i "17i Environment=ETCD_KEY_FILE=$ETCD_KEY_FILE" $TEMPLATE
+       sed -i "20i ExecStartPre=/bin/mkdir -p /var/run/calico" $TEMPLATE
+       sed -i -e "21s#.*#ExecStart=/usr/bin/rkt run --inherit-env --stage1-from-dir=stage1-fly.aci --volume=var-run-calico,kind=host,source=/var/run/calico --volume=modules,kind=host,source=/lib/modules,readOnly=false --mount=volume=modules,target=/lib/modules --volume=dns,kind=host,source=/etc/resolv.conf,readOnly=true --volume=etcd-tls-certs,kind=host,source=$ETCD_CERT_ROOT_DIR,readOnly=true --mount=volume=dns,target=/etc/resolv.conf --mount=volume=etcd-tls-certs,target=/etc/ssl/etcd --mount=volume=var-run-calico,target=/var/run/calico --trust-keys-from-https quay.io/calico/node:v0.19.0#g" $TEMPLATE
+       fi
     fi
 
     local TEMPLATE=/etc/kubernetes/worker-kubeconfig.yaml
@@ -275,6 +301,9 @@
 FLANNELD_IFACE=$ADVERTISE_IP
 FLANNELD_ETCD_ENDPOINTS=$ETCD_ENDPOINTS
 EOF
+    if [ "$ETCD_CLIENT_CERT_AUTH" = true ]; then
+    printf "FLANNELD_ETCD_CAFILE=$ETCD_TRUSTED_CA_FILE\nFLANNELD_ETCD_CERTFILE=$ETCD_CERT_FILE\nFLANNELD_ETCD_KEYFILE=$ETCD_KEY_FILE\n" >> $TEMPLATE
+    fi
     fi
 
     local TEMPLATE=/etc/systemd/system/flanneld.service.d/40-ExecStartPre-symlink.conf.conf
@@ -333,6 +362,11 @@
     }
 }
 EOF
+        if [ "$ETCD_CLIENT_CERT_AUTH" = true ]; then
+       sed -i "7i \     \"etcd_key_file\": \"$ETCD_KEY_FILE\"," $TEMPLATE
+       sed -i "8i \     \"etcd_cert_file\": \"$ETCD_CERT_FILE\"," $TEMPLATE
+       sed -i "9i \     \"etcd_ca_cert_file\": \"$ETCD_TRUSTED_CA_FILE\"," $TEMPLATE
+        fi
     fi
 
     local TEMPLATE=/etc/kubernetes/cni/net.d/10-flannel.conf
